src/contexts/globalLayout/GlobalNavbar/CollectionSaveButtonWithCaption.tsx

```diff
@@ -0,0 +1,155 @@
+import { useCallback, useMemo, useRef, useState } from 'react';
+import styled from 'styled-components';
+
+import { Button } from '~/components/core/Button/Button';
+import colors from '~/components/core/colors';
+import { HStack, VStack } from '~/components/core/Spacer/Stack';
+import { BaseM, TitleDiatypeL } from '~/components/core/Text/Text';
+import { TextAreaWithCharCount } from '~/components/core/TextArea/TextArea';
+import transitions, {
+  ANIMATED_COMPONENT_TRANSITION_MS,
+  ANIMATED_COMPONENT_TRANSLATION_PIXELS_SMALL,
+} from '~/components/core/transitions';
+import CloseIcon from '~/icons/CloseIcon';
+
+type Props = {
+  disabled?: boolean;
+  hasUnsavedChange?: boolean;
+  label?: string;
+  onSave: (caption: string) => Promise<void>;
+};
+
+export function CollectionSaveButtonWithCaption({
+  disabled,
+  hasUnsavedChange,
+  label = 'Save',
+  onSave,
+}: Props) {
+  const [isShowPopup, setIsShowPopup] = useState(false);
+  const [caption, setCaption] = useState('');
+  const deactivateHoverCardTimeoutRef = useRef<ReturnType<typeof setTimeout> | null>(null);
+
+  const handleOpenCaption = useCallback(() => {
+    if (deactivateHoverCardTimeoutRef.current) {
+      clearTimeout(deactivateHoverCardTimeoutRef.current);
+    }
+    setIsShowPopup(true);
+  }, []);
+
+  const handleCloseCaption = useCallback(() => {
+    deactivateHoverCardTimeoutRef.current = setTimeout(
+      () => setIsShowPopup(false),
+      ANIMATED_COMPONENT_TRANSITION_MS
+    );
+  }, []);
+
+  const handleCaptionChange = useCallback((event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setCaption(event.target.value);
+  }, []);
+
+  const handleSubmit = useCallback(async () => {
+    setIsLoading(true);
+    await onSave(caption);
+    handleCloseCaption();
+    setIsLoading(false);
+  }, [caption, handleCloseCaption, onSave]);
+
+  const isDisabledonSave = useMemo(() => {
+    return caption.length > 600;
+  }, [caption]);
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  return (
+    <StyledConfirmationContainer>
```

Comment:
the button length looks like it should be smaller in the mockup

![image](https://user-images.githubusercontent.com/12162433/200096678-6f083f94-0411-41fe-9a6e-eda9c4606fcf.png)

Reply:
Fixed!